standard/ERCs/erc-6900.md

```diff
@@ -471,72 +470,55 @@ An account can have the same validation module installed more than once.
-
-The function `installModule` accepts 3 parameters: the address of the module to install, the Keccak-256 hash of the module's manifest, ABI-encoded data to pass to the module's `onInstall` callback.
-
-The function MUST retrieve the module's manifest by calling `moduleManifest()` using `staticcall`.
-
-The function MUST perform the following preliminary checks:
+During validation installation, the account MUST correctly set flags and other fields based on the incoming data provided by users.
 
-- Revert if the module has already been installed on the modular account.
-- Revert if the module does not implement ERC-165 or does not support the `IModule` interface.
-- Revert if `manifestHash` does not match the computed Keccak-256 hash of the module's returned manifest. This prevents installation of modules that attempt to install a different module configuration than the one that was approved by the client.
-- Revert if any address in `dependencies` does not support the interface at its matching index in the manifest's `dependencyInterfaceIds`, or if the two array lengths do not match, or if any of the dependencies are not already installed on the modular account.
+- the account MUST install all pre validation hooks required by the user and call `onInstall` with the user-provided data on the hook module to initialize the states.
+- the account MUST install all permission hooks required by the user and call `onInstall` with the user-provided data on the hook module to initialize the states.
+- the account MUST add all selectors required by the user that the validation can validate.
+- the account MUST set `isGlobal` and `isSignatureValidation` flags as required.
+- the account MUST call `onInstall` on the validation module to initialize the states.
+- the account MUST emit `ValidationInstalled` as defined in the interface for all installed validations.
 
-The function MUST record the manifest hash and dependencies that were used for the module's installation. Each dependency's record MUST also be updated to reflect that it has a new dependent. These records MUST be used to ensure calls to `uninstallModule` are comprehensive and undo all edited configuration state from installation. The mechanism by which these records are stored and validated is up to the implementation.
+During validation uninstallation, the account MUST correctly clear flags and other fields based on the incoming data provided by users.
 
-The function MUST store the module's permitted function selectors, permitted external calls, and whether it can spend the account's native tokens, to be able to validate calls to `executeFromModule` and `executeFromModuleExternal`.
+- the account MUST clear all flags for the validation function, including `isGlobal` and `isSignatureValidation`.
+- the account MUST remomve all hooks and clear hook module states by calling `onUninstall` with the user-provided data for each hook, including both pre validation hooks and permission hooks.
```

Comment:
```suggestion
- the account MUST remomve all hooks and MAY clear hook module states by calling `onUninstall` with the user-provided data for each hook, including both pre validation hooks and permission hooks.
```

Reply:
Added user intent wording, similar to execution installation behavior.